### ISSUE TITLE
str() doesn't work as expected

### ISSUE THREAD

**Issue by aronchick** [python-semver/python-semver#232]
When I parse a version string into a ```SemVer``` dict, I would expect ```str(semver_variable)``` to return a concatenated string. 

Is there a best practice here? I'd really like to store my variable as a SemVer, but I have to print it often, so having a simple pattern for doing so would be great.

**Comment by tomschr** [comment]
Hi @aronchick 

concatenating a dict directly isn't possible. You can use the `semver.format_version` function like this:

```python
>>> v1 = semver.parse("1.2.3")
>>> semver.format_version(**v1)
'1.2.3'
```

Would that help?

**Comment by aronchick** [comment]
Yep, that's perfect! 

Is it possible to add a clearer method for doing this?

**Comment by tomschr** [comment]
What do you have in mind with "clearer method"? Do you mean the name of the function? Or something different?

Currently, we are discussing to deprecate certain functions for the next major release. For details see #229. It's still under discussion, but depending on the outcome, this function could also be affected.

**Comment by tomschr** [comment]
Not sure if you are aware of another alternative way of doing (almost) the same. You can use the `VersionInfo` class. For example:

```python
>>> v2 = semver.VersionInfo.parse("2.3.4")
>>> str(v2)
'2.3.4'
```

See also the section [Creating a Version](https://python-semver.readthedocs.io/en/latest/usage.html#creating-a-version) in the semver documentation.

**Comment by aronchick** [comment]
so, my methodology is to do the parse once, and then treat it like an object.

In this case, my flow would be (forgive the hacky code):
```
def __init__(self, version):
	__version = semver.parse(version)

def print_version():
	return str(__version) 

a = my_object('0.0.1')
print(f"This object's version is {my_object.print_version()}")
```
or some equivalent.

I can put whatever I want into the accessor method (obviously), but i didn't even know about the 'format_version' method. And, mentally, it doesn't feel right to ".parse()" again - i've created the object. I was just expecting that this object would be able to render itself as a joined string on its own.

**Comment by tomschr** [comment]
Ahh, I see what you mean.

I think a more pythonic way would be avoid the method  `print_version` and to replace it with the magic method `__str__`. This would make the code a bit simpler:

```python
def __init__(self, version):
    __version = semver.parse(version)

def __str__(self):
    return str(self.__version)

# outside the class:
a = my_object('0.0.1')
print(f"This object's version is {a}")
```

This would implicitly call the magic method `__str__`.

However, I'm not sure why you don't use the `semver.VersionInfo` class directly. Probably the above code was just an excerpt and contains a bit more, right?

**Comment by aronchick** [comment]
OH ok, wow. I get it now. I just thought ```semver``` was the object that I wanted to store, not ```semver.VersionInfo```. That was just confusing to me!

Yes, ```VersionInfo``` behaves exactly as expected - it's just strange to me because I would have expected ```semver.parse(<value>)``` to result in the object rather than ```semver.VersionInfo.parse(<value>)```

**Comment by tomschr** [comment]
No problem. :smile: 

Yes, it can be confusing. There are mostly two ways to deal with version information: the object oriented way (with `semver.VersionInfo`) and the "module level functions" way with deal with builtin objects (dicts and `semver.parse`).

I referenced issue #229 which deals exactly with this: to deprecated these "module level functions" so we only have the `semver.VersionInfo` object. IMHO, this would make the project a bit easier to understand and adheres more to the Zen of Python: "There should be one-- and preferably only one --obvious way to do it."

**Comment by scls19fr** [comment]
@tomschr you are 100% allowed to follow this path to improve python-semver!
Could you summarize what is locking in #229 ?

**Comment by tomschr** [comment]
@scls19fr thanks, will do. 👍

@aronchick can we close this issue? It seems, we could solve your question, right? 🙂

**Comment by aronchick** [comment]
Yep! Close away - thanks so much for your help and responsiveness!
